WooCommerce/Classes/Authentication/Epilogue/StorePickerViewController.swift

```diff
@@ -362,13 +398,18 @@ private extension StorePickerViewController {
     /// Re-initializes the Login Flow, forcing a logout. This may be required if the WordPress.com Account has no Stores available.
     ///
     func restartAuthentication() {
-        guard ServiceLocator.stores.needsDefaultStore, let navigationController = navigationController else {
+        guard ServiceLocator.stores.needsDefaultStore else {
             return
         }
 
-        ServiceLocator.stores.deauthenticate()
-
         let loginViewController = ServiceLocator.authenticationManager.loginForWordPressDotCom()
+
+        guard let navigationController = navigationController else {
+            assertionFailure("Navigation error: one of the login / logout states is not correctly handling navigation. No navigationController found.")
```

Comment:
I like this, at least we'd get warned when testing locally that we are in an inconsistent state